src/app/components/TopicHandler/TopicHandler.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import { useParams } from 'react-router-dom'
+import TopicView from '../../pages/TopicView/TopicView'
+
+import type { Topic, Need } from '../../types/types'
+
+type TopicHandlerProps = {
+  topics: Topic[]
+  onNeedUpvote: (
+    topicId: string
+  ) => (needId: string) => (updatedVotes: number) => void
+  onNeedSubmit: (topicId: string) => (need: Need) => void
+}
+
+export default function TopicHandler({
+  topics,
+  onNeedUpvote,
+  onNeedSubmit,
+}: TopicHandlerProps): JSX.Element {
+  const params = useParams()
```

Comment:
You can destructure the params as well!
```suggestion
  const { topicId } = useParams()
```
But then you have a naming douplication the line below, how sad :D